charts/virtual-kubelet-for-aks/templates/secrets.yaml

```diff
@@ -4,6 +4,16 @@ metadata:
   name: {{ template "fullname" . }}
 type: Opaque
 data:
-  cert.pem: {{ (default "TUlTU0lORw==" .Values.env.apiserverCert) | quote }}
-  key.pem: {{ (default "TUlTU0lORw==" .Values.env.apiserverKey) | quote }}
-  clientSecret: {{ default "" .Values.env.azureClientKey | b64enc | quote }}
+  {{- if (not .Values.env.apiserverCert) and (not .Values.env.apiserverKey) }}
+  {{- $ca := genCA "virtual-kubelet-ca" 3650 }}
+  {{- $cn := printf "%s-virtual-kubelet-apiserver" .Release.Name }}
+  {{- $altName1 := printf "%s-virtual-kubelet-apiserver.%s" .Release.Name .Release.Namespace }}
+  {{- $altName2 := printf "%s-cvirtual-kubelet-apiserver.%s.svc" .Release.Name .Release.Namespace }}
+  {{- $cert := genSignedCert $cn nil (list $altName1 $altName2) 3650 $ca }}
+  cert.pem: {{ b64enc $cert.Cert }}
+  key.pem: {{ b64enc $cert.Key }}
+  {{ else }}
+  cert.pem: {{ quote .Values.env.apiserverCert }}
```

Comment:
shall we error out if any of them is empty here? since in L7, you only generate the cert when both of them are empty.

Reply:
The direction in the other PR was to allow things to fall through and let the http listener report the error. I'm good either way, I did this to follow the behavior from that discussion, but I don't have a strong opinion either way. Your call.

Reply:
Sounds fair. we can leave it as it is.